### PR TITLE
fix nx-on-wake daemon

### PR DESCRIPTION
**Pull request by mewlips**
The problem is nx-on-wake executes 'on-wake' on every mount/umount events except when sdcard is not mounted.
I think this is a bug.

### CODE REVIEW

**Comment by vasile-gh** [comment]
@mewlips I need to better understand what you want to do - what is the
behaviour you want to see?

The reason I am running the event on every mount is because there's no
userland means to detect when the camera wakes up - I am using the mount as
a proxy to the real wake up because the SD card is unmounted when the
camera goes to sleep.

So again, what is it that you want to actually see?

rgds

**Comment by mewlips** [comment]
When I hacking the camera, i need to mount/umounts partitions.
for example, when i execute 'mount -o remount,rw /' command on telnet, nx-on-wake executes the nx-patch's on-wake. Then telnet process kllied by on-wake script.

And, I need to run some appplication in chroot environment. At this time, I need to mount /sys and /proc to chroot's root with bind option. There is no reason to execute 'on-wake' again.

So, I modified nx-on-wake to run 'on-wake' if and only if sdcard mount event occurs.

**Comment by vasile-gh** [comment]
@mewlips the current implementation only launches on-wake when the SD card is mounted. It does nothing when the card is unmounted.

From what I can see your code basically replaces the functionality that already exists.

**Comment by mewlips** [comment]
I modified two part of the original code.

First, isSdMounted() function, this is removal of possible overflow of buf when file read.
maybe, this dose not change the action of nx-on-wake.

Second, I added sdMounted variable for tracking the sdcard mount/umount state.
When sdcard unmounted, sdMounted set to zero. This occurs when power off by switch.
(or cold boot from hibernation image, at this time sdMounted is zero)
When poll function return and sdcard is mounted, nx-on-wake launches on-wake.
Then sdMounted set to 1.
After the wake up, mount/umount events should be ignored execpt sdcard unmount event.
The original code launches on-wake on any events of mount/umount.

**Comment by vasile-gh** [comment]
ok. regarding your point 1 (overflow) I was quite aware of this but the target cameras do not have a large mounts file, and I preferred to do a read in one go to gain a few miliseconds :-). But I could accept your code since it does indeed do a cleaner job (at the cost of more complex algo)

regarding point 2 have you tried to see what happens when you pull the card? the camera shuts down. In addition if you read attentively my code you will see that the event is triggered only on SD card mount - that is where the strstr() comes into play in my code.

So, bottom line, I still do not see why the pull should be merged except part one but even for that one, I would prefer to not change anything because I do not want to tempt users to reinstall the mod when there's no need.

Thoughts?

**Comment by mewlips** [comment]
Yes. I compiled nx-on-wake.

Before the installing, I cleaned my camera by updating the original firmware.
Then, I copied the scripts for enable telnet and nx-on-wake files into the sdcard.
After boot into the factory mode, I copied the nx-on-wake files (nx-on-wake, nx-on-wake.service) to root parition and I removed info.tg on sdcard.

Then, I followed below document.
https://github.com/ottokiksmaler/nx500_nx1_modding/blob/master/doc/Update_hibernation_image.md
My camera made sucessfully new hibernation image, and worked I intend to do.

However, I did not tested the original nx-on-wake installation process.
So, more test is required before the distributing to others.
